### DISCUSSION TITLE
Scroll interpolation

### DISCUSSION THREAD

**Discussion by effespwnz** [AvaloniaUI/Avalonia#12945] [General]
Right now when you are using mouse scroll wheel ScrollView scrolls content without any interpolation. It is just instantly moves to next section. It makes working with text pretty hard and overall experience with scroll feels clanky. 

https://github.com/AvaloniaUI/Avalonia/assets/7274123/bd4673b2-5756-4694-a341-9f9fc65abb08

Expected behavior for scroll view is something like presented here. Some interpolation when scroll goes to next snap point.

https://github.com/AvaloniaUI/Avalonia/assets/7274123/2035e258-589e-4053-8144-32b723c41f5c

Video recorded with 60 fps on 60fps monitor. On 120-144Hz monitor difference in scrolling feels worse. There are no problem with gestures. They are working good. I am using Avalonia 11.0.4 version.

**Comment by timunie** [comment]
Now I see what you're talking about. Yeah our TextBox is not yet virtualized which leads to bad performance having a huge amount of text to render. I'd suggest to try AvaloniaEdit for now. 

We haven an open issue to track here: https://github.com/AvaloniaUI/Avalonia/issues/12108

**Reply by effespwnz** [reply]
This is not about FPS drop because there are many objects to render. It is because scroll in avalonia works with a single fire event OnPointerWheelChanged(), which just calculates new position and moves view and ScrollBar thumb. And because it is single fire event it happens instantly. What I am suggesting is interpolated scroll which happens over time so scroll for mouse wheel consist of multiple steps like StartScroll UpdateScroll EndScroll. Or make multiple scroll mods with float as constant multiplier. 0 interpolation time makes scroll instant (same as it works now) and higher values make scrolling reach target position over time.

**Reply by dbriard** [reply]
I just comment to track this topic as I also would be interested by having an animated scroll. Something like ScrollTo/By but with an animation (duration and easing parameters?). For my case, it would be to scroll page by page in a smooth way.

**Reply by rabbitism** [reply]
> I just comment to track this topic as I also would be interested by having an animated scroll. Something like ScrollTo/By but with an animation (duration and easing parameters?). For my case, it would be to scroll page by page in a smooth way.

maybe you can check this video and the mentioned project in description. 

【Avalonia 我想要平滑：循环列表-哔哩哔哩】 https://b23.tv/t5mIbtM

**Comment by timunie** [comment]
Got it working once (for some reason it does only trigger smooth scroll once 🤷 )

```xml
<Style Selector="ScrollViewer">
    <Setter Property="Transitions">
        <Transitions>
            <VectorTransition Property="Offset" Duration="0:0:2" />
        </Transitions>
    </Setter>
</Style>
```

![Animation](https://github.com/AvaloniaUI/Avalonia/assets/47110241/35be37fe-cab4-4bd2-95db-3bd3e54bf625)

@jmacato I guess you are experienced with animations, any idea why it doesn't work several times?
